spopt/tests/test_spenc.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import libpysal
+import geopandas as gpd
+from .. import Spenc
+import unittest
+
+
+class TestSpenc(unittest.TestCase):
+
+
+    def setUp(self):
+        self.pth = libpysal.examples.get_path("mexicojoin.shp")
+        self.mexico = gpd.read_file(self.pth)
+        self.mexico["count"] = 1
+        self.attrs_name = [f"PCGDP{year}" for year in range(1950, 2010, 10)]
+        self.w = libpysal.weights.Queen.from_dataframe(self.mexico)
+        np.random.seed(123456)
+        self.model = Spenc(gdf=self.mexico, w=self.w, attrs_name=self.attrs_name)
```

Comment:
Try adding the random state as a keyword to `Spenc()` that is then passed into `SPENC()`